sled-agent/src/instance_manager.rs

```diff
@@ -727,12 +726,21 @@ impl InstanceManagerRunner {
         tx: oneshot::Sender<Result<ZoneBundleMetadata, BundleError>>,
         name: &str,
     ) -> Result<(), BundleError> {
-        let Some(instance) = self
-            .jobs
-            .iter()
-            .find(|(propolis_id, _)| name == propolis_zone_name(propolis_id))
-            .map(|entry| entry.1)
-        else {
+        // A well-formed Propolis zone name must consist of
+        // `PROPOLIS_ZONE_PREFIX` and the Propolis ID. If the prefix is not
+        // present or the Propolis ID portion of the supplied zone name isn't
+        // parseable as a UUID, there is no Propolis zone with the specified
+        // name to capture into a bundle, so return a `NoSuchZone` error.
+        let vmm_id: PropolisUuid = name
+            .strip_prefix(PROPOLIS_ZONE_PREFIX)
+            .ok_or_else(|| BundleError::NoSuchZone { name: name.to_string() })
+            .and_then(|uuid_str| {
+                uuid_str.parse::<PropolisUuid>().map_err(|_| {
+                    BundleError::NoSuchZone { name: name.to_string() }
+                })
+            })?;
```

Comment:
Super nitpicky, not a big deal: since we discard the parse error, I think it would be a bit simpler to instead convert the result to an `Option` using `Result::ok()`, rather than turning the `Option` into a `Result` and then mapping a different `Result`-returning function over it and throwing away _that_ error. Like this:
```suggestion
            .strip_prefix(PROPOLIS_ZONE_PREFIX)
            .and_then(|uuid_str| uuid_str.parse::<PropolisUuid>().ok())
            .ok_or_else(|| BundleError::NoSuchZone { name: name.to_string() })?;
```